ZombieWar/src/main/java/com/csc422/zombiewar/characters/Teacher.java

```diff
@@ -1,12 +1,22 @@
 package com.csc422.zombiewar.characters;
 
+import com.csc422.zombiewar.weapons.Weapon;
+
 public class Teacher extends Survivor
 {
     private int objID = 0;
+    private Weapon weapon;
+    
+    public Teacher(int ID, Weapon weapon)
+    {
+        super(/*health*/50,/*attack*/5);
+        this.objID = ID;
+        this.weapon = weapon;
```

Comment:
Instead of creating the Weapon variable in the, I think you need to pass the Weapon up to the Survivor class by passing the weapon into super()